community/kernel/src/main/java/org/neo4j/kernel/impl/api/DataIntegrityValidatingStatementOperations.java

```diff
@@ -227,22 +230,26 @@ public void constraintDrop( KernelStatement state, RelationshipPropertyConstrain
         schemaWriteDelegate.constraintDrop( state, constraint );
     }
 
-    private void checkIndexExistence( KernelStatement state, OperationContext context, int labelId, int propertyKey )
+    private void checkIndexExistence( KernelStatement state, OperationContext context,
+            NodePropertyDescriptor descriptor )
             throws AlreadyIndexedException, AlreadyConstrainedException
     {
-        for ( IndexDescriptor descriptor : loop( schemaReadDelegate.indexesGetForLabel( state, labelId ) ) )
+        //TODO: Consider using schemaReadDelegate.indexGetForLabelAndPropertyKey(state, indexDescriptor)
```

Comment:
This pattern of looking up be label first and then by properties is used in many places in the code. I thought it better to address those in a separate refactoring, so that the current PR does not go too far out of scope. I added the TODO as a reminder for that work.